flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/split/PulsarPartitionSplitReaderBase.java

```diff
@@ -196,28 +196,42 @@ public void close() {
     }
 
     @Nullable
-    protected abstract Message<byte[]> pollMessage(Duration timeout)
+    protected abstract Message<?> pollMessage(Duration timeout)
             throws ExecutionException, InterruptedException, PulsarClientException;
 
-    protected abstract void finishedPollMessage(Message<byte[]> message);
+    protected abstract void finishedPollMessage(Message<?> message);
 
-    protected abstract void startConsumer(PulsarPartitionSplit split, Consumer<byte[]> consumer);
+    protected abstract void startConsumer(PulsarPartitionSplit split, Consumer<?> consumer);
+
+    //    protected abstract void startConsumer(PulsarPartitionSplit split, Consumer<OUT> consumer,
+    // Schema<OUT> schema);
 
     // --------------------------- Helper Methods -----------------------------
 
     protected boolean isNotWakeup() {
         return !wakeup.get();
     }
 
+    // TODO change javadoc description
+
     /** Create a specified {@link Consumer} by the given split information. */
-    protected Consumer<byte[]> createPulsarConsumer(PulsarPartitionSplit split) {
+    protected Consumer<?> createPulsarConsumer(PulsarPartitionSplit split) {
         return createPulsarConsumer(split.getPartition());
     }
 
-    /** Create a specified {@link Consumer} by the given topic partition. */
-    protected Consumer<byte[]> createPulsarConsumer(TopicPartition partition) {
-        ConsumerBuilder<byte[]> consumerBuilder =
-                createConsumerBuilder(pulsarClient, Schema.BYTES, configuration);
+    // TODO can we not use exception as a control flow ? We can return null if there is no
+    // corresponding pulsar schema ?
+    protected Consumer<?> createPulsarConsumer(TopicPartition partition) {
+        try {
+            return createPulsarConsumer(partition, deserializationSchema.schema());
+        } catch (UnsupportedOperationException exception) {
+            return createPulsarConsumer(partition, Schema.BYTES);
```

Comment:
I remember there is a NONE schema type, can we use Schema.BYTES here ? Add a test case for it

Reply:
I don't think we should implement logic like this. I would add an extra commit to your PR.

Reply:
Would you mind elaborate why we shouldn't implement logic like this ? Thanks ~